### PR TITLE
Skipif no lzma ne

### PR DESCRIPTION
**Pull request by WillAyd**
- [ ] progress #18190
- [X] tests added / passed
- [X] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/tests/frame/test_query_eval.py [hunk 1]
```diff
@@ -370,27 +361,25 @@ def to_series(mi, level):
                 raise AssertionError("object must be a Series or Index")
 
     def test_raise_on_panel_with_multiindex(self, parser, engine):
-        tm.skip_if_no_ne()
```
Comment (WillAyd) [inline pandas/tests/frame/test_query_eval.py]: Minor bug before where this wasn't passing ``engine`` as an argument to the skip_if function, so this test was always being skipped (even for non-``numexpr`` engines)

File: pandas/tests/frame/test_query_eval.py [hunk 2]
```diff
@@ -370,27 +361,25 @@ def to_series(mi, level):
...
         p = tm.makePanel(7)
         p.items = tm.makeCustomIndex(len(p.items), nlevels=2)
         with pytest.raises(NotImplementedError):
             pd.eval('p + 1', parser=parser, engine=engine)
 
     def test_raise_on_panel4d_with_multiindex(self, parser, engine):
-        tm.skip_if_no_ne()
```
Comment (WillAyd) [inline pandas/tests/frame/test_query_eval.py]: Same bug as above

File: pandas/tests/frame/test_query_eval.py [hunk 3]
```diff
@@ -1037,63 +1017,29 @@ def test_query_string_scalar_variable(self, parser, engine):
...
-        cls.engine = 'numexpr'
-        cls.parser = 'python'
-        tm.skip_if_no_ne(cls.engine)
-
-
-class TestDataFrameEvalPythonPandas(TestDataFrameEvalNumExprPandas):
-
-    @classmethod
-    def setup_class(cls):
-        super(TestDataFrameEvalPythonPandas, cls).setup_class()
```
Comment (WillAyd) [inline pandas/tests/frame/test_query_eval.py]: The call to the super ``setup_class`` method inadvertently skipped this class of tests on systems that did not have ``numexpr`` even though that isn't required for these. Fixed with commit

**Comment by jreback** [comment]
thanks @WillAyd keep em coming!

**Pull request merged**